brainiak/fcma/classifier.py

```diff
@@ -0,0 +1,278 @@
+
+.. [Wang2015] Full correlation matrix analysis (FCMA): An unbiased method for
+   task-related functional connectivity",
+   Yida Wang, Jonathan D Cohen, Kai Li, Nicholas B Turk-Browne.
+   Journal of Neuroscience Methods, 2015.
+"""
+
+# Authors: Yida Wang
+# (Intel Labs), 2016
+
+import numpy as np
+import time
+from sklearn.base import BaseEstimator
+import sklearn
+from . import fcma_extension
+from . import cython_blas as blas
+import logging
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "Classifier",
+]
+
+
+class Classifier(BaseEstimator):
+    """Correlation-based classification component of FCMA
+
+    Parameters
+    ----------
+
+    clf: class
+        The classifier used, normally a classifier class of sklearn
+
+    epochs_per_subj: int, default 0
+        The number of epochs of each subject
+        within-subject normalization will be performed during
+        classifier training if epochs_per_subj is specified
+        default 0 means no within-subject normalization
+
+    training_data: 2D numpy array in shape [num_samples, num_features]
+        default None
+        training_data is None except clf is SVM.SVC with precomputed kernel,
+        in which case training data is needed to compute
+        the similarity vector for each sample to be classified
+
+    num_voxels: int
+        The number of voxels per brain used in this classifier
+        this is defined by the applied mask, normally the top voxels
+        selected by FCMA voxel selection
+        num_voxels must be consistent in both training and classification
+
+    num_samples: int
+        The number of samples of the training set
+    """
+    def __init__(self,
+                 clf,
+                 epochs_per_subj=0):
+        self.clf = clf
+        self.epochs_per_subj = epochs_per_subj
+        self.training_data = None
+        self.num_voxels = -1
+        self.num_samples = -1
+        return
+
+    def fit(self, X, y):
+        """ use correlation data to train a model
+
+        the input data X is activity data filtered by top voxels
+        and prepared for correlation computation.
+        X needs to be first converted to correlation,
+        and then normalized within subject
+        if more than one sample in one subject,
+        and then fit to a model defined by self.clf.
+
+        Parameters
+        ----------
+        X: a list of numpy array in shape [nun_TRs, num_voxels]
+            assuming all elements of X has the same num_voxels value
+        y: labels, len(X) equals len(Y)
+
+        Returns
+        -------
+        self: return the object itself
+        """
+        time1 = time.time()
+        assert len(X) == len(y), \
+            'the number of samples does not match the number labels'
+        num_samples = len(X)
+        num_voxels = X[0].shape[1]  # see assumption above
+        self.num_voxels = num_voxels
+        self.num_samples = num_samples
```

Comment:
You should add an `_` to attributes set based on the data:
http://scikit-learn.org/stable/developers/contributing.html#estimated-attributes